### ISSUE TITLE
Publishing build results errors from harbormaster (Parameter 'name' has invalid type. Expected type 'string', got type 'null'.)

### ISSUE THREAD

**Issue by joshma** [uber/phabricator-jenkins-plugin#148]
I'm seeing this at the end of my build:

```
00:10:03.822 [phabricator:process-build-result] Publishing unit results to Harbormaster for 3735 tests.
00:10:03.823 [phabricator:process-build-result] Sending build result to Harbormaster with PHID PHID-HMBT-tvek7qdtxa63kumyxs5d, success: false
00:10:05.307 [phabricator:send-harbormaster-result] Error from Harbormaster: Parameter 'name' has invalid type. Expected type 'string', got type 'null'.
00:10:05.308 [phabricator:send-harbormaster-result] Unable to post to Harbormaster
00:10:05.308 [phabricator:send-harbormaster-result] Error sending Harbormaster unit results, trying again without unit data (you may have an old Phabricator?).
```

I'm guessing an updated version of Harbormaster started requiring `name` and this plugin isn't passing it? I'll see if I can narrow it down but thought I'd report it first.

**Comment by sectioneight** [comment]
Cool, thanks for the report. I'll look at the new Harbormaster API at work
on Monday and figure out what we need to do.

**Comment by joshma** [comment]
Thanks @sectioneight ! Happy to test things for you :)

**Comment by sectioneight** [comment]
Hmm, it doesn't look like anything has changed (at least according to the public documentation: https://secure.phabricator.com/conduit/method/harbormaster.sendmessage/)

We're already sending a "name" parameter which is set to the test name.

Is it possible your test names are null in your junit report? Would you mind posting a (possibly sanitized) version of the test results that your job is trying to publish?

We're running a recent stable version of Phabricator at Uber and haven't run into this yet, and I don't see anything obvious that's changed on the Harbormaster side either.

Did this just start happening for you recently?

**Comment by joshma** [comment]
Ooh interesting. We have two projects, and one of them is working. The second one has names being passed, but there is this peculiar line:
```
<testcase classname="ChaiFailsLoudly" name="null" time="0"/>
```

I'm going to try without that test and see if that's the problem.

**Comment by joshma** [comment]
Yep, confirmed - we had a test named "null" (we were testing a chai matcher `null()`) and I renamed it to "expects null" and it now works. So the issue seems to be that the plugin is converting a string `"null"` into a JSON value `null`

**Comment by sectioneight** [comment]
That's... lame. We're not actually doing any JSON parsing ourselves, so this must be happening in the JSON library we're using. I'll see if there's a valid workaround.

**Comment by joshma** [comment]
Cool - it's not blocking us anymore, since we should've had better test names anyways, but probably something nice to fix.
